pkgs/stdenv/darwin/override-sdk.nix

```diff
@@ -0,0 +1,412 @@
+  replacePropagatedPackages =
+    newPackages: pkg:
+    let
+      propagatedInputs = lib.attrValues {
+        inherit (pkg)
+          depsBuildBuildPropagated
+          propagatedNativeBuildInputs
+          depsBuildTargetPropagated
+          depsHostHostPropagated
+          propagatedBuildInputs
+          depsTargetTargetPropagated
+          ;
+      };
+
+      env = {
+        inherit (pkg) outputs;
+
+        # Old dependencies are replaced with new ones via a sed script. It is assumed
+        # that `|` will not appear in a store path.
+        dependencies = lib.pipe propagatedInputs [
+          lib.flatten
+          (lib.concatMapStrings (
+            dep:
+            let
+              replacement = getReplacement newPackages dep;
+            in
+            lib.optionalString (dep != replacement) "s|${dep}|${replacement}|g;"
+          ))
+        ];
+
+        passAsFile = [ "dependencies" ];
+      };
+    in
+    # Only remap the package’s propagated inputs if there are any and if any of them
+    # had packages remapped (with frameworks or proxy packages).
+    if
+      lib.isDerivation pkg && lib.any (inputs: inputs != [ ]) propagatedInputs && env.dependencies != ""
+    then
+      mkStub pkg.name env
+    else
+      pkg;
+
+  propagatedInputs = [
+    "depsBuildBuildPropagated"
+    "propagatedNativeBuildInputs"
+    "depsBuildTargetPropagated"
+    "depsHostHostPropagated"
+    "propagatedBuildInputs"
+    "depsTargetTargetPropagated"
+  ];
+
+  # Gets all propagated inputs in a package. This does not recurse.
+  getPropagatedInputs =
+    pkg:
+    lib.concatMap (input: pkg.${input} or [ ]) [
+      "depsBuildBuildPropagated"
+      "propagatedNativeBuildInputs"
+      "depsBuildTargetPropagated"
+      "depsHostHostPropagated"
+      "propagatedBuildInputs"
+      "depsTargetTargetPropagated"
+    ];
+
+  # Gets all propagated dependencies in a package.
+  getPropagatedDependencies =
+    pkgs:
+    lib.genericClosure {
+      startSet = map (pkg: {
+        key = pkg.outPath;
+        deps = getPropagatedInputs pkg;
+      }) pkgs;
+      operator =
+        { key, deps }:
+        map (dep: {
+          key = dep.outPath;
+          deps = getPropagatedInputs dep;
+        }) deps;
+    };
+
+  # Returns a package mapping based on remapping all propagated packages.
+  #
+  # Implementation note: recursion is avoided because it is costly. Instead, a flat
+  # list of dependencies is obtained then sorted topologically, so that dependencies
+  # can be updated in order (and only once). This produces a new mapping that must
+  # be used to perform the actual updates to the propagated packages.
+  getPackageMapping =
+    baseMapping: input:
+    let
+      dependencies = lib.pipe input [
+        getPropagatedDependencies
+        (lib.toposort (x: y: lib.any (z: x.key == z) y.deps))
+        (lib.getAttr "result")
+        lib.reverseList
```

Comment:
I don’t think that will work. `toposort` is doing a DFS using the comparison function to define which nodes are adjacent to the current node. I’d need to change `getPropagatedDependencies` to return a mapping of dependency to parents, which doesn’t seem worth doing just to avoid a `reverseList`.